syz-ci/syz-ci.go

```diff
@@ -219,7 +219,7 @@ func main() {
 		case <-shutdownPending:
 		case <-updatePending:
 		}
-		kernelBuildSem <- struct{}{} // wait for all current builds
+		buildSem.Wait() // wait for all current builds
```

Comment:
Since you over-generalize the semaphore to support more than 1 parallel build, this now needs to be a new WaitAll method. Or just restrict the semaphore to count 1.

Reply:
Also syz-ci will now exit when bisection jobs are running, right?
I suspect this may lead to a large number of aborted bisection jobs since job latency is O(hours) and we restart every O(hours). And since we retry them only after 3 days, some may be delayed by week+ and cause lots of wasted resources.
We now may also consume more transient VMs. Previously we consumed either VMs for bisection or for a build testing, but now we may consume VMs for bisection and for 2 build tests (1 manager kernel build and 1 patch testing).

Reply:
> We now may also consume more transient VMs. Previously we consumed either VMs for bisection or for a build testing, but now we may consume VMs for bisection and for 2 build tests (1 manager kernel build and 1 patch testing).

That's a good point, though we'd anyway consume more transient VM if we want to do jobs in parallel. We could limit it by a bisection + build/patch test, then the maximum extra is `20+3` instead of `20+3+3`.

Reply:
> Also syz-ci will now exit when bisection jobs are running, right?

Actually, nothing changes in this regard. An update eventually closes the `stop` channel, but it has no effect on individual jobs. Already started jobs continue until they finish, and only then the job loop halts. Until job loop has not returned, syz-ci won't restart as they're connected via a `WaitGroup`.

Reply:
I've added the code to gate `env.Test` calls. Hopefully it can be a reasonable compromise.

Reply:
> > Also syz-ci will now exit when bisection jobs are running, right?
> 
> Actually, nothing changes in this regard. An update eventually closes the `stop` channel, but it has no effect on individual jobs. Already started jobs continue until they finish, and only then the job loop halts. Until job loop has not returned, syz-ci won't restart as they're connected via a `WaitGroup`.

But then isn't the following possible:
 - a bisection job has just started
 - syz-ci starts to exit for an update and closes stop channel (it will wait for just 1 of bisection builds to finish, but not the whole process)
 - all managers will stop, patch testing requests will stop
 - bisection will continue for 8 more hours, but syz-ci will be idle otherwise (no fuzzing, testing requests delayed)
?

Reply:
Hmm, yes, you're right. That's not the desirable behavior.

We could explicitly wait for the jobs to finish instead of waiting for all builds to finish. I've just done this in https://github.com/google/syzkaller/pull/3607/commits/37cabb03efeb12e09bc2597ce41fefe1301b1838.

What do you think?